src/utils/buildConceptHierarchy.ts

```diff
@@ -0,0 +1,31 @@
+import { TFamilyConcept } from "@/types";
+
+export type TFamilyConceptTreeNode = TFamilyConcept & {
+  children: TFamilyConceptTreeNode[];
+};
+
+function buildTree(concept: TFamilyConcept, allConcepts: TFamilyConcept[], visited: Set<string>): TFamilyConceptTreeNode {
+  // Prevent infinite cycles
+  if (visited.has(concept.id)) {
```

Comment:
Question: I'm assuming this is just in case the API response has circular data, rather than any error of our FE processing?

Reply:
Yeh, if the API has given us the same concept somehow as a child of itself, we don't want to crash the app in an infinite loop!